### PR TITLE
Verify contract exists 0.5

### PR DESCRIPTION
**Pull request by i-stam**
similar to SafeERC20

### CODE REVIEW

File: contracts/wallet.sol [hunk 1]
```diff
@@ -711,13 +713,14 @@ contract Wallet is ENSResolvable, Vault, GasTopUpLimit, LoadLimit {
...
     /// @param _value ETH amount in wei
     /// @param _data transaction payload binary
-    function _executeCall(address _to, uint256 _value, bytes memory _data) internal  returns (bool success)
-{
-    // solium-disable-next-line security/no-inline-assembly
-    assembly {
-        success := call(gas, _to, _value, add(_data, 0x20), mload(_data), 0, 0)
+    function _executeCall(address _to, uint256 _value, bytes memory _data) internal  returns (bool success){
+        //check if contract exists
+        require(address(_to).isContract(), "_executeCall: call to non-contract");
```
[the remark this answers is not in the document]
Comment (nostdm) [reply contracts/wallet.sol]: Why did we change this function @i-stam ? Is this from the upgraded contract?

[the remark this answers is not in the document]
Comment (i-stam) [reply contracts/wallet.sol]: this is for safety (suggested by the auditors) check the new version that i pushed

[the remark this answers is not in the document]
Comment (nostdm) [reply contracts/wallet.sol]: OK

**Comment by nostdm** [comment]
https://github.com/tokencard/contracts/issues/402

**Comment by mischat** [comment]
@nostdm am fixing this branch now ...

File: contracts/wallet.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (mischat) [reply contracts/wallet.sol]: @i-stam this doesn't seem right to me ... i am fixing a push. This logic around isContract or not is already covered in the executeTransaction call. I am removing this now.

**Pull request merged**